### PR TITLE
improve completers

### PR DESCRIPTION
**Pull request by jnoortheen**
## Todo

- [x] ~~implement git completer~~ - seems like a bigger task than expected. Maybe some other time. In the meantime one can use bash/fish/carapace

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by Techcable** [comment]
I just tested this and it appears to fix #4652.

It also appears to fix a number of other issues I've had with completers.

In particular automatic quoting and handling of spaces seems much more reliable 😄 

I'll continue to test this and let you know if I encounter any problems!

**Comment by jnoortheen** [comment]
@Techcable you may also want to checkout https://github.com/rsteube/carapace-bin . It is faster than fish completer .

**Comment by Techcable** [comment]
For anyone else who wants to try this out, there's an easy way to get a`.patch` file of this PR. Just append `.patch` to the end of the URL.On our case, this would be https://github.com/xonsh/xonsh/pull/4648.patch - Then GitHub should redirect

Download the patch file with `wget "https://github.com/xonsh/xonsh/pull/4648.patch"`.

Then run `git am 4684.patch`

So the final sequence of commands is:
```shell
# NOTE: You can skip this step if you're already on the correct branch
git fetch upstream
git checkout main
git reset --hard upstream/main 
wget "https://github.com/xonsh/xonsh/pull/4648.patch"
git am 4648.patch
```

It will add the latest version of the PR for your to test out 🎉 .

This process is automatic and works for any GitHub PR (not just this one). It also works for commits too :)

**Review by gforsyth — APPROVED** [review]
This looks great @jnoortheen !  And everything is running smoothly on my end.

File: news/completions-improve.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+**Added:**
+
+* native completer for ``git``, no longer bash is needed for completing git.
```
[the remark this answers is not in the document]
Comment (gforsyth) [reply news/completions-improve.rst]: outdated

**Comment by gforsyth** [comment]
If you want to fix the small docstring nit, go for it, otherwise I think you're good to merge this in.

**Comment by jnoortheen** [comment]
> If you want to fix the small docstring nit, go for it, otherwise I think you're good to merge this in.

Thanks I committed it.

**Pull request merged**